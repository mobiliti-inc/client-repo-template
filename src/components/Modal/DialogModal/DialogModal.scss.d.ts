// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'box-container': string;
  'button-group': string;
  'cancel-button-container': string;
  'fades-in-out': string;
  'main-container': string;
  'save-button-container': string;
  'save-button-container--disabled': string;
  'title': string;
}
declare var cssExports: CssExports;
export = cssExports;
