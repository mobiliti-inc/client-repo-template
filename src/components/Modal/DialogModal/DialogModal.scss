@import "../../../styles/_variables.scss";

.main-container {
  display: flex;
  flex-direction: column;
  width: 80%;
}

.title {
  max-height: 62px;
  max-width: 286px;
  color: $darker-blue;
  font-family: $futura;
  font-size: 22px;
  font-weight: bold;
  letter-spacing: -0.49px;
  line-height: 31px;
  margin: 0 0 20px 0;
  word-wrap: break-word;
}

.box-container {
  display: flex;
  flex-direction: column;
  width: 100%;
}

.button-group {
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-column-gap: 8px;
}

.save-button-container {
  width: 80%;
  opacity: 1;

  &--disabled {
    opacity: 0.6;
  }
}

.cancel-button-container {
  align-self: center;
  height: 25px;
  width: 124px;
  font-family: $futura;
  font-size: 14px;
  font-weight: bold;
  line-height: 23px;
  text-align: center;

  & button {
    color: $blue;
    border: none;
    background-color: white;
    cursor: pointer;

    &:focus {
      outline: none;
    }
  }
}