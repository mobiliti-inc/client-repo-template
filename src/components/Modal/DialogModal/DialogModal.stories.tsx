import { storiesOf } from '@storybook/react';
import { text, boolean } from '@storybook/addon-knobs';
import * as React from 'react';
import { DialogModal } from '../../';

storiesOf('DialogModal', module)
	.add('default', () =>
		<DialogModal
			blueButtonClick={() => { }}
			blueButtonText={text('blueButtonText', "Continue")}
			closeModal={() => { }}
			footer={boolean('footer', true)}
			isBlueButtonDisabled={boolean('isBlueButtonDisabled', false)}
			modalIsVisible={boolean('modalIsVisible', true)}
			modalTitle={text('modalTitle', "Modal title")}
			whiteButtonClick={() => { }}
			whiteButtonText={text('whiteButtonText', "Back")}
		>
			<h2>Modal Body</h2>
		</DialogModal>
	);
