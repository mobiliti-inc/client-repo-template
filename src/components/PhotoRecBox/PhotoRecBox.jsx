import styles from './PhotoRecBox.scss';

const PhotoRecBox = ({
	onBoxClick,
	boxIcon,
	isVertical,
	boxTitle,
	boxDescription,
	boxImage,
	onUploadImageInputChange
}) => (
	<div role="presentation" onClick={onBoxClick} styleName={`rec-box rec-box--${isVertical ? 'vertical' : 'horizontal'}`} >
		{boxIcon &&
		<span styleName={`circle circle--${isVertical ? 'vertical' : 'horizontal'}`}>
			<img src={boxIcon} alt="box icon" />
		</span>
		}

		{<input
			styleName="input-front-license"
			type="file"
			name="front-license"
			id="front-license"
			accept="image/jpeg"
			onChange={onUploadImageInputChange}

		/>}
		{boxImage &&
		<div styleName="box-image-container">
			<img styleName="box-image" src={boxImage} alt="box" />
			<div styleName="check-mark-circle">
				<span styleName="check-mark" />
			</div>
		</div>}
		{boxTitle && <div styleName={`box-title box-title--${isVertical ? 'vertical' : 'horizontal'}`}>{boxTitle}</div>}
		{boxDescription &&
			<div
				styleName={`box-description box-description--${isVertical ? 'vertical' : 'horizontal'}`}
			>{boxDescription}
			</div>}

	</div >);


PhotoRecBox.propTypes = {
	isVertical: PropTypes.bool,
	boxIcon: PropTypes.string,
	boxTitle: PropTypes.string,
	boxDescription: PropTypes.string,
	onBoxClick: PropTypes.func.isRequired,
	boxImage: PropTypes.string,
	onUploadImageInputChange: PropTypes.func
};

PhotoRecBox.defaultProps = {
	isVertical: true,
	boxIcon: '',
	boxTitle: '',
	boxDescription: '',
	boxImage: '',
	onUploadImageInputChange: () => { }
};

export default CSSModules(PhotoRecBox, styles, { allowMultiple: true });

