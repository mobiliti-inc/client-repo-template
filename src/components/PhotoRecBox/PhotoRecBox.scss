@import "~variables";
.rec-box {
    width: 100%;
    border: 1px solid rgba(0, 0, 0, 0.219);
    border-radius: 5px;
    cursor: pointer;
    display: flex;
    align-items: center;
    justify-content: center;
    &--vertical {
        flex-direction: column;
        padding: 40px 0;
    }
    &--horizontal {
        flex-direction: row;
        padding: 20px 0;
        justify-content: flex-start;
    }
}

.circle {
    height: 66px;
    width: 66px;
    border: 2px solid $darker-blue;
    border-radius: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    &--vertical {
        margin-left: 0;
    }
    &--horizontal {
        margin-left: 10%;
    }
}

.box-title {
    width: 144px;
    color: $darker-blue;
    font-family: $futura;
    font-size: 18px;
    font-weight: bold;
    letter-spacing: -0.4px;
    line-height: 20px;
    text-align: center;
    &--vertical {
        padding: 15px 0 0 0;
    }
    &--horizontal {
        padding: 0 0 0 15px;
    }
}

.box-description {
    width: 173px;
    color: $darker-blue;
    font-family: $hanken-sans;
    font-size: 16px;
    line-height: 19px;
    text-align: center;
    &--vertical {
        padding: 5px 5px;
    }
    &--horizontal {
        padding: 0;
    }
}

.box-image-container {
    margin-left: 10%;
    width: 122px;
    height: 91px;
    display: flex;
    justify-content: center;
    align-items: center;
    position: relative;
    box-shadow: $box-shadow--standard;
}

.box-image {
    width: 100%;
    height: 100%;
    border-radius: 4px;
}

.check-mark-circle {
    position: absolute;
    background-color: $green-check;
    color: white;
    height: 26px;
    width: 26px;
    border-radius: 100%;
    top: -13px;
    right: -13px;
    display: flex;
    justify-content: center;
    align-items: center;
}

.check-mark {
    @include check-mark;
    display: inline-block;
    cursor: pointer;
}

.input-front-license {
    display: none;
}