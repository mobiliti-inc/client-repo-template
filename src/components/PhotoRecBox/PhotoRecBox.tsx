import styles from './PhotoRecBox.scss';

type PhotoRecBoxProps = {
	isVertical?: boolean,
	boxIcon?: string,
	boxTitle?: string,
	boxDescription?: string,
	onBoxClick: (...args: any[]) => any,
	boxImage?: string,
	onUploadImageInputChange?: (...args: any[]) => any
};

const PhotoRecBox: React.SFC<PhotoRecBoxProps> = ({
	onBoxClick,
	boxIcon,
	isVertical,
	boxTitle,
	boxDescription,
	boxImage,
	onUploadImageInputChange
}) => (
		<div role="presentation" onClick={onBoxClick} styleName={`rec-box rec-box--${isVertical ? 'vertical' : 'horizontal'}`}>
			{boxIcon && (
				<span styleName={`circle circle--${isVertical ? 'vertical' : 'horizontal'}`}>
					<img src={boxIcon} alt="box icon" />
				</span>
			)}

			{
				<input
					styleName="input-front-license"
					type="file"
					name="front-license"
					id="front-license"
					accept="image/jpeg"
					onChange={onUploadImageInputChange}
				/>
			}
			{boxImage && (
				<div styleName="box-image-container">
					<img styleName="box-image" src={boxImage} alt="box" />
					<div styleName="check-mark-circle">
						<span styleName="check-mark" />
					</div>
				</div>
			)}
			{boxTitle && <div styleName={`box-title box-title--${isVertical ? 'vertical' : 'horizontal'}`}>{boxTitle}</div>}
			{boxDescription && (
				<div styleName={`box-description box-description--${isVertical ? 'vertical' : 'horizontal'}`}>{boxDescription}</div>
			)}
		</div>
	);

// PhotoRecBox.defaultProps = {
// 	isVertical: true,
// 	boxIcon: '',
// 	boxTitle: '',
// 	boxDescription: '',
// 	boxImage: '',
// 	onUploadImageInputChange: () => {}
// };

export default CSSModules(PhotoRecBox, styles, { allowMultiple: true });
