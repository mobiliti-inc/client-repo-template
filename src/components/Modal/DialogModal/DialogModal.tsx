import React from 'react';
// @ts-ignore
import Modal, { EasyModalHeader } from 'easy-modal-react';

import { Button } from '../../';

import './DialogModal.scss';

interface DialogModalProps {
	blueButtonClick?: (...args: any[]) => any;
	whiteButtonClick?: (...args: any[]) => any;
	modalIsVisible?: boolean;
	closeModal: (...args: any[]) => any;
	modalTitle: string;
	whiteButtonText?: string;
	blueButtonText?: string;
	isBlueButtonDisabled?: boolean;
	className?: string;
	footer?: boolean;
}

const DialogModal: React.FC<DialogModalProps> = props => {
	const {
		blueButtonClick,
		modalTitle,
		isBlueButtonDisabled,
		blueButtonText,
		whiteButtonText,
		children,
		whiteButtonClick,
		modalIsVisible,
		closeModal,
		footer
	} = props;

	return (
		<Modal
			open={modalIsVisible}
			onClose={closeModal}
			header={<EasyModalHeader onClose={closeModal} />}
			footer={
				footer ? (
					<div styleName="button-group">
						{blueButtonText && (
							<div
								styleName={`save-button-container ${isBlueButtonDisabled ? 'save-button-container--disabled' : ''}`}
							>
								<Button onClick={blueButtonClick} disabled={isBlueButtonDisabled}>
									{blueButtonText}
								</Button>
							</div>
						)}
						{whiteButtonText && (
							<div styleName="cancel-button-container">
								<button onClick={whiteButtonClick}>{whiteButtonText}</button>
							</div>
						)}
					</div>
				) : null
			}
		>
			<div styleName="title">{modalTitle}</div>
			{children}
		</Modal>
	);
};

export default DialogModal;
