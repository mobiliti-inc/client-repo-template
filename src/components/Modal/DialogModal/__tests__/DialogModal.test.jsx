import DialogModal from '../DialogModal';
import PhotoRecBox from '../../../PhotoRecBox/PhotoRecBox';

describe('DialogModal', () => {
	it('renders', () => {
		const tree = shallow(
			<DialogModal
				modalTitle="test headline"
				blueButtonClick={() => {}}
				whiteButtonClick={() => {}}
				modalIsVisible
				closeModal={() => {}}
				whiteButtonText="test"
				blueButtonText="message test"
				isBlueButtoDisabled
			/>
		);
		matchSnapshot(tree);
	});

	it('renders with children', () => {
		const tree = shallow(
			<DialogModal
				modalTitle="test headline"
				blueButtonClick={() => {}}
				whiteButtonClick={() => {}}
				modalIsVisible
				closeModal={() => {}}
				whiteButtonText="test"
				blueButtonText="message test"
				isBlueButtoDisabled={false}
			>
				<PhotoRecBox boxIcon="test icon" onBoxClick={() => {}} />
			</DialogModal>
		);
		matchSnapshot(tree);
	});
});
