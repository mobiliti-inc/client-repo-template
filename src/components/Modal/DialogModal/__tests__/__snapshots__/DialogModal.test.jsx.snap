// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogModal renders 1`] = `
<t
  className=""
  footer={
    <div
      className="button-group"
    >
      <div
        className="save-button-container save-button-container--disabled"
      >
        <Button
          bordered={false}
          buttonType="standard"
          className=""
          customStyles=""
          disabled={true}
          icon={null}
          inverted={false}
          onClick={[Function]}
          ripple={false}
          showLoader={false}
        >
          message test
        </Button>
      </div>
      <div
        className="cancel-button-container"
      >
        <button
          onClick={[Function]}
        >
          test
        </button>
      </div>
    </div>
  }
  header={
    <t
      onClose={[Function]}
    />
  }
  onClose={[Function]}
  open={true}
>
  <div
    className="title"
  >
    test headline
  </div>
</t>
`;

exports[`DialogModal renders with children 1`] = `
<t
  className=""
  footer={
    <div
      className="button-group"
    >
      <div
        className="save-button-container"
      >
        <Button
          bordered={false}
          buttonType="standard"
          className=""
          customStyles=""
          disabled={false}
          icon={null}
          inverted={false}
          onClick={[Function]}
          ripple={false}
          showLoader={false}
        >
          message test
        </Button>
      </div>
      <div
        className="cancel-button-container"
      >
        <button
          onClick={[Function]}
        >
          test
        </button>
      </div>
    </div>
  }
  header={
    <t
      onClose={[Function]}
    />
  }
  onClose={[Function]}
  open={true}
>
  <div
    className="title"
  >
    test headline
  </div>
  <PhotoRecBox
    boxDescription=""
    boxIcon="test icon"
    boxImage=""
    boxTitle=""
    isVertical={true}
    onBoxClick={[Function]}
    onUploadImageInputChange={[Function]}
  />
</t>
`;
